src/System.Memory/src/System/Buffers/ReadOnlySequenceDebugView.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+
+namespace System.Buffers
+{
+    internal sealed class ReadOnlySequenceDebugView<T>
+    {
+        private readonly T[] _array;
+
+        private readonly ReadOnlySequenceDebugViewSegments _segments;
+
+        public ReadOnlySequenceDebugView(ReadOnlySequence<T> sequence)
+        {
+            _array = sequence.ToArray();
+
+            var segmentCount = 0;
+            foreach (var _ in sequence)
+            {
```

Comment:
Maybe we should have a property on ReadOnlySequence called segment count which returns this.

Reply:
We thought about it. It would be useful but I don't want to expose a property that requires list traversal every time. To make calculation O(1) changes to ROSSegment is required and it's a bit outside of the scope of this PR.